### PR TITLE
Miscellaneous refactoring and small fixes

### LINKED ISSUE #341: [Enhancement] If no semicolons present, use commas for choose
### Description of Problem / Inefficiency:
If I accidentally type `?choose a, b`  instead of `a; b` then it'll not give me what I expect

### Implementation Suggestion:
Check if `;` is present; if so split on that, otherwise split on `,`

### LINKED ISSUE #342: [Enhancement] Update assignable roles
Add `Continuing Studies` as a faculty, `Accounting` as a field (for div)

### PR DESCRIPTION
**Pull request by davidlougheed**
## Description

* Move LaTeX preamble for ?tex to outside the command, to avoid re-defining it every call and make it easier to update
* Moves @le-potate 's feels-like weather calculation into an enclosed function
* Added f-strings in place of `.format()` where useful/nice
* Fixed wording on TEPID status to avoid saying "at least one printer is down" when both are down
* Narrowed exception clauses in some places, or made notes of where it should be done
* Shrank history look-back for image commands to improve performance
* Make image processing functions properly static (because they are)
* Fix ?bac and ?mix errors on blank arguments
* Refactor `auto_incorrect` to reduce pointless lines / move defines out of function
* Refactor ?pyramid and limit the unit size to avoid obnoxious messages / API size limits
* Add some missing english equivalents in reminders and code-ify the replacements
* Remove some useless comments that restated the obvious
* Other misc. code styling stuff - snake case, useless encoding headers, copyright updates
* Closes #341 
* Closes #342 

## Motivation and Context
Trying to bring all our code up to modern Python standards, while fixing various tiny issues encountered over the past little while.

## How Has This Been Tested?
Commands have been tested on my own server.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: cogs/helpers.py [hunk 1]
```diff
@@ -89,66 +89,68 @@ async def exam(self, ctx):
...
-        Data taken from http://weather.gc.ca/city/pages/qc-147_metric_e.html"""
+        """
+        Retrieves current weather conditions.
+        Data taken from http://weather.gc.ca/city/pages/qc-147_metric_e.html
+        """
         def retrieve_string(label):
             return soup.find(
                 "dt", string=label).find_next_sibling().get_text().strip()
 
+        def calculate_feels_like(temp: float, humidity: float, ws_kph: float) \
```
Comment (namemcguffin) [inline cogs/helpers.py:100]: shouldn't this be defined outside the weather command? maybe either as a private static method, or as a function in utils?

File: cogs/helpers.py [hunk 2]
```diff
@@ -89,66 +89,68 @@ async def exam(self, ctx):
 
     @commands.command()
     async def weather(self, ctx):
-        """Retrieves current weather conditions.
-        Data taken from http://weather.gc.ca/city/pages/qc-147_metric_e.html"""
+        """
+        Retrieves current weather conditions.
+        Data taken from http://weather.gc.ca/city/pages/qc-147_metric_e.html
+        """
         def retrieve_string(label):
```
Comment (namemcguffin) [inline cogs/helpers.py:96]: shouldn't this be defined outside the weather command? maybe either as a private static method, or as a function in utils?

File: cogs/helpers.py (hunk 1 above)
Comment (davidlougheed) [reply cogs/helpers.py:100]: sure, private static just above is probably fine

File: cogs/helpers.py (hunk 2 above)
Comment (davidlougheed) [reply cogs/helpers.py:96]: less sure about this one cause its tiny but yeah maybe

**Review by namemcguffin — APPROVED** [review]

**Review by namemcguffin — APPROVED** [review]
finally us undergrads can rise up and have the best role

**Review by namemcguffin — APPROVED** [review]

**Pull request merged**